### PR TITLE
NEW : allow stock management by product

### PR DESCRIPTION
**Pull request by thomas-Ngr**
# NEW : allow stock management by product

When using product stocks, users may want to have some products for which they don't want to manage stocks, for example in case of subcontracting.

This PR:
- adds a field "Stockable product" on product card (editable on update, not on create)
- sets this option to 1 by default
- hides the "stocks" tabs if this option is not set
- does not manage stock movements for a product if this option is not set.

Discussion started on #26184 and #26200

SQL Part : #26190

### CODE REVIEW

File: htdocs/product/class/product.class.php [hunk 1]
```diff
@@ -620,6 +627,21 @@ class Product extends CommonObject
 	 * Service
 	 */
 	const TYPE_SERVICE = 1;
+	/**
+	 * Advanced feature: assembly kit
+	 */
+	const TYPE_ASSEMBLYKIT = 2;
```
Comment (eldy) [inline htdocs/product/class/product.class.php:633]: It seems this is not related to the feature stockable or not.
Thanks to split this other feature into a different PR.

**Comment by thomas-Ngr** [comment]
@eldy I have removed the only faulty line, can you have a look ?

**Comment by thomas-Ngr** [comment]
follow-up in #30230
